sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/BeamKafkaThriftTable.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.meta.provider.kafka;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import org.apache.beam.sdk.io.thrift.ThriftCoder;
+import org.apache.beam.sdk.io.thrift.ThriftSchema;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.SchemaProvider;
+import org.apache.beam.sdk.schemas.transforms.Convert;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.Values;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.apache.thrift.TBase;
+import org.apache.thrift.TFieldIdEnum;
+import org.apache.thrift.protocol.TProtocolFactory;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public class BeamKafkaThriftTable<FieldT extends TFieldIdEnum, T extends TBase<T, FieldT>>
+    extends BeamKafkaTable {
+  private final ThriftCoder<T> thriftCoder;
+  private final TypeDescriptor<T> typeDescriptor;
+
+  public BeamKafkaThriftTable(
+      @NonNull Schema requiredSchema,
+      @NonNull String bootstrapServers,
+      @NonNull List<String> topics,
+      @NonNull Class<T> thriftClass,
+      @NonNull TProtocolFactory protocolFactory) {
+    super(thriftSchema(thriftClass, requiredSchema), bootstrapServers, topics);
+    typeDescriptor = TypeDescriptor.of(thriftClass);
+    thriftCoder = ThriftCoder.of(thriftClass, protocolFactory);
+  }
+
+  private static Schema thriftSchema(
+      @NonNull Class<?> thriftClass, @NonNull Schema requiredSchema) {
+    @SuppressWarnings("nullness")
+    final @NonNull Schema thriftSchema =
+        ThriftSchema.provider().schemaFor(TypeDescriptor.of(thriftClass));
+    if (!requiredSchema.equivalent(thriftSchema)) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Given message schema: '%s'%n"
+                  + "does not match schema inferred from thrift class.%n"
+                  + "Thrift class: '%s'%n"
+                  + "Inferred schema: '%s'",
+              requiredSchema, thriftClass.getName(), thriftSchema));
+    }
+    return thriftSchema;
+  }
+
+  @Override
+  protected PTransform<PCollection<KV<byte[], byte[]>>, PCollection<Row>> getPTransformForInput() {
+    final @NonNull SchemaProvider schemaProvider = ThriftSchema.provider();
+    return new PTransform<PCollection<KV<byte[], byte[]>>, PCollection<Row>>() {
+      @Override
+      @SuppressWarnings("nullness")
```

Comment:
I'd try to get rid of this nullness suppression. Usually if (var != null) {} else { throw NullPointerException("var was null") is unfortunatelly needed to achieve that (that's my experience, maybe there is some cleaner way that I don't know about)

Reply:
I'll give it a try.